app/src/main/java/com/lagradost/cloudstream3/extractors/Rabbitstream.kt

```diff
@@ -0,0 +1,167 @@
+package com.lagradost.cloudstream3.extractors
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.lagradost.cloudstream3.SubtitleFile
+import com.lagradost.cloudstream3.app
+import com.lagradost.cloudstream3.base64DecodeArray
+import com.lagradost.cloudstream3.utils.AppUtils
+import com.lagradost.cloudstream3.utils.AppUtils.parseJson
+import com.lagradost.cloudstream3.utils.ExtractorApi
+import com.lagradost.cloudstream3.utils.ExtractorLink
+import com.lagradost.cloudstream3.utils.M3u8Helper
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import java.util.Objects
+import javax.crypto.Cipher
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+class Megacloud : Rabbitstream() {
+    override val name = "Megacloud"
+    override val mainUrl = "https://megacloud.tv"
+    override val embed = "embed-2/ajax/e-1"
+    override val key = "https://raw.githubusercontent.com/enimax-anime/key/e6/key.txt"
+}
+
+class Dokicloud : Rabbitstream() {
+    override val name = "Dokicloud"
+    override val mainUrl = "https://dokicloud.one"
+}
+
+open class Rabbitstream : ExtractorApi() {
```

Comment:
It looks like you are using something related to https://github.com/enimax-anime/enimax please add some sort of link to that in the top of the file along with the licence of enimax